packages/shared/src/components/modals/NewSourceModal.tsx

```diff
@@ -62,6 +64,10 @@ export default function NewSourceModal(props: ModalProps): ReactElement {
   const { user, loginState, showLogin } = useContext(AuthContext);
   const loginTrigger = AuthTriggers.SubmitNewSource;
   const { onRequestClose } = props;
+  const [isDismissed, setIsDismissed] = usePersistentContext(
+    DISMISS_PERMISSION_BANNER,
+    false,
+  );
```

Comment:
So does this only work for the `NewSourceModal` permissions?
Or for all permission prompts?

Reply:
This is for all prompts. Sorry I had merged already, there was a split second where the comment wasn't shown yet.

Reply:
Ah I see, maybe a hook would be better as there are 2 places it's shared now, could introduce some weird edge-cases.

Reply:
The behavior will be the same even if we introduce a hook as we instantiate it in two places. I think this should be fine. Introducing a hook would probably look like below:

```
const useDismissedPermission = () => usePersistentContext(
    DISMISS_PERMISSION_BANNER,
    false,
  );
  
// implementation
const [isDismissed, setIsDismissed] = useDismissedPermission();
```

Should I create a hotfix PR for refactor?

Reply:
Hmm, nah let's leave it.
If we introduce yet another one, let's add a "usePermissions" hook or something.

Reply:
I agree, something we can generalize permissions 🚢 
Thanks for the consideration! 🚀